tools/mtouch/mtouch.Designer.cs

```diff
@@ -0,0 +1,2090 @@
+﻿//------------------------------------------------------------------------------
+// <auto-generated>
+//     This code was generated by a tool.
+//     Runtime Version:4.0.30319.42000
+//
+//     Changes to this file may cause incorrect behavior and will be lost if
+//     the code is regenerated.
+// </auto-generated>
+//------------------------------------------------------------------------------
+
+namespace Errors {
+    using System;
+    using System.Reflection;
+    
+    
+    [System.CodeDom.Compiler.GeneratedCodeAttribute("System.Resources.Tools.StronglyTypedResourceBuilder", "4.0.0.0")]
+    [System.Diagnostics.DebuggerNonUserCodeAttribute()]
+    [System.Runtime.CompilerServices.CompilerGeneratedAttribute()]
+    public class mtouch {
```

Comment:
This must be an `internal` class, since it ends up in the class library we ship (Xamarin.iOS.dll). 

Also we end up adding this whole file to the class library, when most of it is not referenced. I think we should come up with a different solution for the code that's included the class library, one that doesn't end up adding all these properties to the library.

One solution that comes to mind is to add extra metadata to the entries in the resx to mark those error messages used in Xamarin.iOS.dll, and implement a custom generator to generate the corresponding code.

Reply:
hmm... maybe that's a step too far ? i.e. does it make sense to translate *runtime* errors ? my understanding is that we were asked to localize build errors/warnings. c.c. @chamons  

Even if those errors share the same number space (no dupes) they are not `mtouch/mmp` errors, they are runtime errors - and others are not localized (e.g. all exception messages).

If we want them localized then it might be better to do this into a separate PR since the process will be different - at least for building (e.g. some extra tool like @rolfbjarne suggested). Strings are large and we do not want to include unneeded ones into final applications, let alone several translations for [un]needed ones :-)

Reply:
@spouliot the thing is that the code is shared between mtouch/mmp and the runtime. That works fine while using string constants, but it gets more complicated now. A potential solution would be an adaptation of the previous solution I suggested, where the custom generator I proposed would write out properties that just return the corresponding string constant (so no need to adapt the src/ build to consume the resx).

Reply:
wouldn't a secondary resx file that just contains errors  ONLY used by src/ObjCRuntime/Registrar.c and src/ObjCRuntime/RuntimeOptions.cs  (and other runtime code) work? There might be some tiny duplication of errors but it's a lot simpler then writing an additional tool.

Reply:
ok, I thought it also contained the parts that are runtime only...

we do have a `#if MTOUCH || MMP` in a few places... so we could have the platform assemblies ship with a subset (generated or manual)

not ideal... but, at least for now, the autogenerated file is part of the repo :|

Reply:
A secondary resx file isn't a bad idea, but the split could be different: one with errors for mtouch+mmp, and one with errors for mtouch+mmp+runtime. That way there's no duplication. We'd still have to fix the src/ build to embed the resx in the assembly, and we'd have to check the size increase (if any) to see if it's acceptable.

Long-term it might be a good idea to move all user-visible strings from the assemblies themselves and into external resource files, which might help decrease the final executable size (unless the AOT compiler already puts string data in the .aotdata files).

@spouliot what do you think?